infor-design/enterprise-wc
# 130 - Implement IDS App Menu

**Explain the details for making this change. What existing problem does the pull request solve?**
This PR adds the IDS App Menu component to the WebComponents library, along with supporting features/libraries/etc:
- Adds component code, tests, docs, etc. for App Menu
- Adds a new component, `IdsDrawer` that serves as the base for components that can "slide out" from the viewport edges.  Added this for use with the App Menu, and future components like Action Sheet
- Adds the capability for nesting and selection to Accordions
- Added color variants to the Accordion that are used by the App Menu
- ...among other minor changes to the Accordion/Toolbar to facilitate use inside the App Menu

**Related github/jira issue (required)**:
Closes #130 

**Steps necessary to review your pull request (required)**:
Pull/Build/Run, then:

_Test App Menu:_
- Open http://localhost:4300/ids-app-menu
- Use the trigger button to activate the IdsAppMenu
- Dismiss the app menu both by clicking outside, or by pressing Escape
- Focus accordion headers within the App Menu and use the Up/Down arrows to navigate.  Navigation should work with different levels of nesting, and should loop around after reaching either the top or bottom of the accordion.
- Expand/Collapse of accordion panes should occur as expected and animate smoothly
- When clicking an accordion header, a "selected" state with bolded text should appear
- Open dev tools and change the IdsContainer element's language prop to `ar`.  RTL should work on all App Menu elements.

_Test IdsDrawer:_
- Open http://localhost:4300/ids-drawer
- Use the trigger buttons to activate the Drawer components, and click outside of them to dismiss.  (Note the "Action Sheet" is not a fully-working Action Sheet component yet, and just shows its style)

_Test Nested Accordion:_
- Open http://localhost:4300/ids-accordion/nested.html
- Try Expand/Collapse and keyboard navigation on both accordion types and ensure they work as expected.

In general, all tests/lints/etc should be passing.

**Included in this Pull Request**:
